core/integration/gpu_test.go

```diff
@@ -0,0 +1,77 @@
+package core
+
+import (
+	"context"
+	"os/exec"
+	"regexp"
+	"strings"
+	"testing"
+
+	"dagger.io/dagger"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	cudaImage = "nvidia/cuda:11.7.1-base-ubuntu20.04"
+)
+
+var (
+	// uuidRegex is used to match GPU UUID in nvidia-smi output:
+	uuidRegex = regexp.MustCompile(`GPU-\w{8}-\w{4}-\w{4}-\w{4}-\w{12}`)
+)
+
+func TestGPUAccess(t *testing.T) {
+	ctx := context.Background()
+	c, err := dagger.Connect(ctx)
+	require.NoError(t, err)
+	defer c.Close()
+
+	// Query nvidia-smi on the host:
+	hostNvidiaCmd := exec.Command("nvidia-smi", "-L")
+	hostNvidiaOutput, err := hostNvidiaCmd.Output()
+	hostNvidiaOutputStr := string(hostNvidiaOutput)
+	require.NoError(t, err)
+	require.NotEmpty(t, hostNvidiaOutput)
+
+	// Query the same on the Dagger container and compare output:
+	ctr := c.Container().From(cudaImage)
+	contents, err := ctr.
+		// WithGPU(dagger.ContainerWithGPUOpts{Devices: "GPU-5d8950fe-17a6-2fa7-9baa-afa83bba0e2b"}).
+		WithGPU(dagger.ContainerWithGPUOpts{Devices: "all"}).
+		WithExec([]string{"nvidia-smi", "-L"}).
```

Comment:
This just lists the GPUs to make sure they are visible right? If so, that's great for a basic test, but have you verified programs that actually utilize the GPU work?

I'm guessing there's probably some python ML libraries that could be run fairly in a `WithExec`, would be good to have that test too.

Reply:
@sipsma I covered this a few weeks ago with a test called `TestGPUAccessWithPython`.